client/client.go

```diff
@@ -406,21 +428,37 @@ func (c *Client) Sync(ctx context.Context, options ...SyncOptions) error {
 	return nil
 }
 
-// Watch subscribes to events on a given documentIDs.
+// Subscribe subscribes to events on a given document.
+func (c *Client) Subscribe(
+	doc *document.Document,
+) (<-chan WatchResponse, context.CancelFunc, error) {
+	attachment, ok := c.attachments[doc.Key()]
+	if !ok {
+		return nil, nil, ErrDocumentNotAttached
+	}
+
+	if attachment.isSubscribed.Load() {
+		return nil, nil, ErrAlreadySubscribed
+	}
+	attachment.isSubscribed.Store(true)
```

Comment:
I think this employs the "check-then-act" pattern, where it first checks the state and then changes it. I suspect that there could be a race condition.

How about introducing CompareAndSet, which can perform state comparison and setting atomically?

https://pkg.go.dev/sync/atomic#Bool.CompareAndSwap

Reply:
Oops, this will definitely create race condition. Thank you for your feedback :)